lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2977,6 +3002,11 @@ void ByteCodeGenerator::EmitOneFunction(ParseNode *pnode)
                     }
                 });
             }
+        }
+
+        if (pnode->sxFnc.HasNonSimpleParameterList() || !funcInfo->IsBodyAndParamScopeMerged())
```

Comment:
Do we need to check IsBodyAndParamScopeMerged here? Can that be false if HasNonSimpleParameterList is also false? Is this just for ForceSplitScope?

Reply:
I don't think we can have split scope without a non-simple parameter list unless ForceSplitScope is turned on. I moved our code to emit the special symbol storage above, though, because we don't really care if the scopes are merged. We just need to know if the parameter list is non-simple.